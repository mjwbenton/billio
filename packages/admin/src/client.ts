import { ApolloClient, InMemoryCache } from "@apollo/client";

const CLIENT = new ApolloClient({
  uri: "http://localhost:4000/",
  cache: new InMemoryCache(),
  defaultOptions: {
    query: {
      fetchPolicy: "network-only",
    },
  },
});

export default CLIENT;
