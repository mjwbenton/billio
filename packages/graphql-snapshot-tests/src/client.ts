import { ApolloClient, InMemoryCache, HttpLink } from "@apollo/client";
import fetch from "cross-fetch";

const ENDPOINT = process.env.BILLIO_ENDPOINT;

if (!ENDPOINT) {
  throw new Error("Please configure BILLIO_ENDPOINT for tests");
}

const CLIENT = new ApolloClient({
  cache: new InMemoryCache(),
  link: new HttpLink({
    uri: ENDPOINT,
    fetch,
  }),
});

export default CLIENT;
