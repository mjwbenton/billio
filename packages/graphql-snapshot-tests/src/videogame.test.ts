import client from "./client";
import { gql } from "@apollo/client";

const ITEM_MATCHER = {
  id: expect.any(String),
};

let TEST_VIDEO_GAME: string = "";

test("can import external video game", async () => {
  const { data } = await client.mutate({
    mutation: gql`
      mutation Test_ImportOri {
        firstImport: importExternalVideoGame(
          id: "igdb:136149"
          shelfId: Completed
        ) {
          id
          title
          shelf {
            id
          }
        }
        secondImport: importExternalVideoGame(
          id: "igdb:19456"
          shelfId: Playing
        ) {
          id
          title
          shelf {
            id
          }
        }
      }
    `,
  });
  expect(data).toMatchSnapshot({
    firstImport: ITEM_MATCHER,
    secondImport: ITEM_MATCHER,
  });
  TEST_VIDEO_GAME = data.firstImport.id;
});

test("can query single video game", async () => {
  const { data } = await client.query({
    query: gql`
      query Test_QuerySingleGame($id: ID!) {
        videoGame(id: $id) {
          id
          title
          shelf {
            id
            name
          }
          rating
          platforms {
            id
            name
          }
        }
      }
    `,
    variables: {
      id: TEST_VIDEO_GAME,
    },
  });
  expect(data).toMatchSnapshot({
    videoGame: ITEM_MATCHER,
  });
});

test("can fetch second page of video games", async () => {
  const query = gql`
    query Test_VideoGamesPagination($after: ID) {
      videoGames(first: 1, after: $after) {
        items {
          title
        }
        hasNextPage
        nextPageCursor
      }
    }
  `;
  const { data: first } = await client.query({
    query,
  });
  expect(first).toMatchSnapshot();

  const { data: second } = await client.query({
    query,
    variables: {
      after: first.videoGames.nextPageCursor,
    },
  });
  expect(second).toMatchSnapshot();
});

test("can fetch video games by shelf", async () => {
  const { data } = await client.query({
    query: gql`
      {
        videoGameShelf(id: Completed) {
          name
          items(first: 1) {
            items {
              title
            }
          }
        }
      }
    `,
  });
  expect(data).toMatchSnapshot();
});

test("can search for external video games", async () => {
  const { data } = await client.query({
    query: gql`
      {
        searchExternalVideoGame(term: "The Last of Us") {
          id
          title
        }
      }
    `,
  });
  expect(data).toMatchSnapshot();
});

test("can mutate title on video game", async () => {
  const randomTitle = Math.random()
    .toString(36)
    .replace(/[^a-z]+/g, "")
    .substr(0, 5);
  const { data } = await client.mutate({
    mutation: gql`
      mutation Test_MutateTitle($id: ID!, $title: String!) {
        updateVideoGame(item: { id: $id, title: $title }) {
          title
        }
      }
    `,
    variables: {
      id: TEST_VIDEO_GAME,
      title: randomTitle,
    },
  });
  expect(data.updateVideoGame.title).toEqual(randomTitle);
});

test("can move video game between shelves", async () => {
  const { data } = await client.mutate({
    mutation: gql`
      mutation Test_MoveShelf($id: ID!) {
        updateVideoGame(item: { id: $id, shelfId: Played }) {
          shelf {
            id
            name
          }
        }
      }
    `,
    variables: {
      id: TEST_VIDEO_GAME,
    },
  });
  expect(data).toMatchSnapshot();
});
