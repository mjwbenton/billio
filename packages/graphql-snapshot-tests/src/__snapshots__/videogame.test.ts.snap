// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can fetch second page of video games 1`] = `
Object {
  "videoGames": Object {
    "__typename": "VideoGamePage",
    "hasNextPage": true,
    "items": Array [
      Object {
        "__typename": "VideoGame",
        "title": "Ori and the Blind Forest: Definitive Edition",
      },
    ],
    "nextPageCursor": "eyJjb3VudFNvRmFyIjoxLCJsYXN0S2V5Ijp7InR5cGU6aWQiOiJWaWRlb0dhbWU6NGFjNGIzYjYtNmE2NS00Y2UzLWE1ZGEtMzk0ZmRhYmM2NmViIiwidHlwZSI6IlZpZGVvR2FtZSIsInVwZGF0ZWRBdDp0eXBlOmlkIjoiMTYyNjI2NzIwNzY1NTpWaWRlb0dhbWU6NGFjNGIzYjYtNmE2NS00Y2UzLWE1ZGEtMzk0ZmRhYmM2NmViIn19",
  },
}
`;

exports[`can fetch second page of video games 2`] = `
Object {
  "videoGames": Object {
    "__typename": "VideoGamePage",
    "hasNextPage": true,
    "items": Array [
      Object {
        "__typename": "VideoGame",
        "title": "hyzfi",
      },
    ],
    "nextPageCursor": "eyJjb3VudFNvRmFyIjoyLCJsYXN0S2V5Ijp7InR5cGU6aWQiOiJWaWRlb0dhbWU6M2M1MGNiNmMtMTYyZi00MjYxLThjZmMtMDcxNWI3OWQ0MmUxIiwidHlwZSI6IlZpZGVvR2FtZSIsInVwZGF0ZWRBdDp0eXBlOmlkIjoiMTYyNjI2NzIxMjE1NTpWaWRlb0dhbWU6M2M1MGNiNmMtMTYyZi00MjYxLThjZmMtMDcxNWI3OWQ0MmUxIn19",
  },
}
`;

exports[`can fetch video games by shelf 1`] = `
Object {
  "videoGameShelf": Object {
    "__typename": "VideoGameShelf",
    "items": Object {
      "__typename": "VideoGamePage",
      "items": Array [
        Object {
          "__typename": "VideoGame",
          "title": "Ori and the Will of the Wisps: Collector's Edition",
        },
      ],
    },
    "name": "Completed",
  },
}
`;

exports[`can import external video game 1`] = `
Object {
  "firstImport": Object {
    "__typename": "VideoGame",
    "id": Any<String>,
    "shelf": Object {
      "__typename": "VideoGameShelf",
      "id": "Completed",
    },
    "title": "Ori and the Will of the Wisps: Collector's Edition",
  },
  "secondImport": Object {
    "__typename": "VideoGame",
    "id": Any<String>,
    "shelf": Object {
      "__typename": "VideoGameShelf",
      "id": "Playing",
    },
    "title": "Ori and the Blind Forest: Definitive Edition",
  },
}
`;

exports[`can move video game between shelves 1`] = `
Object {
  "updateVideoGame": Object {
    "__typename": "VideoGame",
    "shelf": Object {
      "__typename": "VideoGameShelf",
      "id": "Played",
      "name": "Played",
    },
  },
}
`;

exports[`can query single video game 1`] = `
Object {
  "videoGame": Object {
    "__typename": "VideoGame",
    "id": Any<String>,
    "platforms": Array [],
    "rating": null,
    "shelf": Object {
      "__typename": "VideoGameShelf",
      "id": "Completed",
      "name": "Completed",
    },
    "title": "Ori and the Will of the Wisps: Collector's Edition",
  },
}
`;

exports[`can search for external video games 1`] = `
Object {
  "searchExternalVideoGame": Array [
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:26192",
      "title": "The Last of Us Part II",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:1009",
      "title": "The Last of Us",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:6036",
      "title": "The Last of Us Remastered",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:11790",
      "title": "The Last of Us: Left Behind",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:101462",
      "title": "The Last of Us Ellie Edition",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:102081",
      "title": "The Last of Us Joel Edition",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:102082",
      "title": "The Last of Us Survival Edition",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:102083",
      "title": "The Last of Us Post-Pandemic Edition",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:135986",
      "title": "The Last of Us Part II: Ellie Edition",
    },
    Object {
      "__typename": "ExternalVideoGame",
      "id": "igdb:135956",
      "title": "The Last of Us Part II: Special Edition",
    },
  ],
}
`;
